exonum-java-binding-core/rust/src/utils/services.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{collections::HashSet, fs::File, io::Read, path::Path};
+
+use toml;
+
+pub const PATH_TO_SERVICES_TO_ENABLE: &str = "ejb_app_services.toml";
+pub const CONFIGURATION_SERVICE: &str = "configuration";
+pub const BTC_ANCHORING_SERVICE: &str = "btc-anchoring";
+pub const TIME_SERVICE: &str = "time";
+pub const EJB_SERVICE: &str = "ejb-service";
+
+#[derive(Serialize, Deserialize)]
+struct ServicesToEnable {
+    services: HashSet<String>,
+}
+
+type Result<T> = std::result::Result<T, Box<std::error::Error>>;
+
+/// Loads service names from a specific TOML configuration file
+pub fn load_enabled_services<P: AsRef<Path>>(path: P) -> Result<HashSet<String>> {
+    let mut file = File::open(path)?;
+    let mut toml = String::new();
+    file.read_to_string(&mut toml)?;
+    let ServicesToEnable { services } =
+        toml::from_str(&toml).expect("Invalid list of services to enable");
+    Ok(services)
+}
+
+/// Determines whether particular service name is defined in the services configuration file.
+/// TODO: For the moment we're just checking for the service's name presence in the configuration
+/// file. As soon as we have dynamic services implemented this checking should happen in runtime.
+pub fn is_service_enabled(service_name: &str) -> bool {
+    is_service_enabled_in_file(service_name, PATH_TO_SERVICES_TO_ENABLE)
```

Comment:
wow. such java. so generic. much enterprise :D

Reply:
Well, otherwise unit-testing it with temporary files (like /some/where/deeeeply/in/tmp/app_services_random_blah_blah.toml_still_random) becomes a little more complicated :) And on the other hand, I'm almost sure that compiler will inline that function in production code, so... :)